util/app/discovery/discovery.go

```diff
@@ -165,7 +165,12 @@ func cmpSupports(ctx context.Context, pluginSockFilePath, appPath, repoPath, fil
 		return nil, nil, false
 	}
 
-	isSupported, isDiscoveryEnabled, err := matchRepositoryCMP(ctx, appPath, repoPath, cmpClient, env, tarExcludedGlobs)
+	// if plugin name is specified, lets return the client directly
+	if namedPlugin {
```

Comment:
I think we still only want to do this if discovery is not enabled, since the CMP author might be using discovery as a sort of safety or even security mechanism.

We probably need a pre-flight request to get `isDiscoveryEnabled` without sending the whole repo. That's a somewhat more involved change, since it'll introduce a new API endpoint.

Reply:
Make a lot of sense, let me update the MR
Thanks a lot for your feedback!

Reply:
What do you think @crenshaw-dev ?

```proto
message CheckRequest {
    // appName refers to the ArgoCD Application name
    string appName = 1;
}

message CheckResponse {
    bool isDiscoveryEnabled = 1;
}


...

// CheckPluginConfiguration is a pre-flight request  to check the plugin configuration
    // without sending the whole repo.
    rpc CheckPluginConfiguration(CheckRequest) returns (CheckResponse) {
    }
```

Reply:
I think the rough idea makes sense. I'm not sure it would need to be a streaming request. If I understand correctly, we wouldn't need to send any streaming contents to the CMP.

Reply:
Yep I was thinking the same, I ve mimic other operations but no strong opinion about stream or unary op

Reply:
Done! Let me know wdyt!
Thanks for all your support :)

Reply:
Not sure how I have to fix this error 
https://github.com/argoproj/argo-cd/actions/runs/9495734280/job/26168614018?pr=18053

Any clue?